hypothesis/h-periodic
# Modernise h-periodic

Fixes https://github.com/hypothesis/h-periodic/issues/10

Make h-periodic the same as all our other apps as far as:

* Dependabot and pip-compile pinned requrements and requirements management
* Latest version of Celery
* pyenv and tox based dev env with `make dev`, `make format`, `make lint`, `make sure`, etc.
* Black, isort and PyLint
* Python 3.6.9 in both the dev and production (Docker) envs (previously the Python version was unspecified)
* Use the standard Python 3.6.9 Docker image, not Alpine
* Gunicorn (with auto-reloading in dev)
* GitHub Actions
* Use Pyramid instead of Flask (and the latest version instead of an unspecified version)

Testing in the development environment
--------------------------------------

Run `make dev`.

http://localhost:8080/_status should return a 200 OK response if h's `make services` is running, and a 500 if h's `make services` is not running (you can take down h's services with `make services args='down'` in the `h` dir).

While h-periodic's `make dev` is running if you also run `make dev` in h you should see log lines (in h) that look like this when the periodic tasks run:

    Task h.tasks.cleanup.purge_deleted_annotations[e9c...9fb] succeeded in 0.001...36s: None

Unfortunately the tasks only run once an hour so you may have to wait!

You can also try `make format`, `make lint` and `make sure`.

Testing in Docker
-----------------

Run `make docker` then `make run-docker`.

http://localhost:8080/_status should return a 200 OK response if h's `make services` is running, and a 500 if h's `make services` is not running (you can take down h's services with `make services args='down'` in the `h` dir).

While `make run-docker` is running if you run `make dev` in h you should see log lines (in h) that look like this when the periodic tasks run:

    Task h.tasks.cleanup.purge_deleted_annotations[e9c...9fb] succeeded in 0.001...36s: None

Unfortunately the tasks only run once an hour so you may have to wait!
